applications/box.md

```diff
@@ -0,0 +1,213 @@
+
+As we spoke to our developer and non-developer friends, it became apparent that even many developers were not aware of these projects and many non-developers did not have the technical knowledge to set them up on their computers, even though it was just a few lines of command.
+
+Besides, there were not many applications that could connect to them or their UX were not as appealing as those developed by big-tech.
+
+We thought a framework that combines easy-to-use backend APIs, easy enough for an average front-end developer, and a pre-setup hardware, plug and play for a non-technical person, can fill this gap and bring decentralized web to everyone’s home.
+
+### Project Details 
+We expect the teams to already have a solid idea about your project's expected final state. Therefore, we ask the teams to submit (with urls or IPFS file CIDs included) at following :
+
+* Mockups/designs of any UI components: We are working on our first DApp that works with IPFS network and box. It is a replacement for "Google Photos" that let the users to store their photos on a "box" and on IPFS. Github project is here: [https://github.com/functionland/photos](https://github.com/functionland/photos)
+
+Below is a GIF demo from "Photos" application, which is a react native app, as the first demo of how an application interacts with "box"
+
+![https://dev.to/fx/google-photos-open-source-alternative-with-react-native-80c](https://raw.githubusercontent.com/functionland/blog/main/photos-intro-dev-to/photos.gif)
+
+* Data models / API specifications of the core functionality
+
+![https://dev.to/fx/google-photos-open-source-alternative-with-react-native-80c](https://raw.githubusercontent.com/functionland/functionland.github.io/main/static/images/box-diagram.jpg)
+
+* An overview of the technology stack to be used
+
+"box" is a collection of integrations for creating a server software that runs js-ipfs in a Node.js environment to serve application data. We use libp2p to create the required protocols for front-end developers to interface with the back-end "box" node, which is running on ipfs. Developers can send commands such as fetch files, upload, or access database. This is achieved by importing a single library that exposes the "box" functionalities to them, @functionland/graph (inspired by [Microsoft Graph workflow](https://docs.microsoft.com/en-us/graph/use-the-api)). These commands are then transferred to "box" node using libp2p. On the back-end the commands are interpreted by the "box" and are sent to js-ipfs or database library to get executed. Database library is using IPLD, specifically DAG-JSON/DAG-JOSE. Multiple "box"es are synchronized and accessible using ipfs-cluster.
+
+Finally, as an availability guarantee for sensitive data, "box" provides the capability to pin data with Crust network, to give them a peace of mind over data safety. They have the option to backup their data to Filecoin network to preserve their important information from data loss.
```

Comment:
```suggestion
Finally, as an availability guarantee for sensitive data, "box" provides the capability to pin data with Crust network, to give them a peace of mind over data safety. They also have the option to backup their data to Filecoin network to preserve their important information from data loss. 

The use cases for Crust and Filecoin are different and box connects the user to both networks. Per our research Filecoin primary use case is a blockchain-style guarantee for high longevity that file is stored forever on the blockchain, on the other hand, Crust is about high availability to ensure user that they can have the files at anytime for any duration they want as quick as possible. The box as a package can be used on any hardware(e.g. laptop) and connects the user and their box applications to crust network for file pinning, which gives the user a way to try the package. In addition, Crust plans for "adding file sharing links" and "privacy storage", might enable us to partner with crust to provide box-as-a-service to users to enable them try the features without owning any hardware. We give both options as the use cases are different.
```

Reply:
This is updated in response to: https://github.com/crustio/Crust-Grants-Program/pull/39/files#r669246083